### PR TITLE
Release of Azure Networking Network-2018-06-01 (api-version 2018-04-01)

### PR DESCRIPTION
**Pull request by MikhailTryakhov**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/loadBalancer.json [hunk 1]
```diff
@@ -1206,11 +1206,10 @@
         },
         "protocol": {
           "type": "string",
-          "description": "The protocol of the end point. Possible values are: 'Http', 'Tcp', or 'Https'. If 'Tcp' is specified, a received ACK is required for the probe to be successful. If 'Http' or 'Https' is specified, a 200 OK response from the specifies URI is required for the probe to be successful.",
+          "description": "The protocol of the end point. Possible values are: 'Http' or 'Tcp'. If 'Tcp' is specified, a received ACK is required for the probe to be successful. If 'Http' is specified, a 200 OK response from the specifies URI is required for the probe to be successful.",
           "enum": [
             "Http",
-            "Tcp",
-            "Https"
```
Comment (MikhailTryakhov) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/loadBalancer.json]: breaking

File: specification/network/resource-manager/readme.md [hunk 2]
```diff
@@ -31,37 +31,6 @@ openapi-type: arm
 tag: package-2018-04
 ```
 
-### Tag: package-2018-05
```
Comment (MikhailTryakhov) [inline specification/network/resource-manager/readme.md]: No need to revert these changes. package version set to ```2018-04``` is enough

Comment (MikhailTryakhov) [reply specification/network/resource-manager/readme.md]: Fixed

File: specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/networkInterface.json [hunk 3]
```diff
@@ -669,6 +669,7 @@
     "NetworkInterfacePropertiesFormat": {
       "properties": {
         "virtualMachine": {
+          "readOnly": true,
```
Comment (MikhailTryakhov) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/networkInterface.json]: What is this from?

File: specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/examples/VmssVmNetworkInterfaceList.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "parameters" : {
-    "api-version" : "2017-03-30",
+    "api-version": "2017-03-30",
```
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/examples/VmssVmNetworkInterfaceList.json]: Shouldn't all versions be `2018-04-01` now? Please update all the examples here accordingly

File: specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/applicationGateway.json [hunk 5]
```diff
@@ -0,0 +1,2142 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "NetworkManagementClient",
+    "description": "The Microsoft Azure Network management API provides a RESTful set of web services that interact with Microsoft Azure Networks service to manage your network resources. The API has entities that capture the relationship between an end user and the Microsoft Azure Networks service.",
+    "version": "2018-06-01"
```
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/applicationGateway.json]: If changes are being ported from 06 to 04, there shouldn't be a `2018-06-01` directory in this PR

**Review by dsgouda — CHANGES_REQUESTED** [review]
Left a couple of comments.
@MikhailTryakhov why is there a `2018-06-01` directory in this PR?

File: specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/examples/VmssVmNetworkInterfaceList.json (hunk 4 above)
Comment (MikhailTryakhov) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/examples/VmssVmNetworkInterfaceList.json]: @dsgouda  as I know we shouldn't update VMSS

Comment (MikhailTryakhov) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/examples/VmssVmNetworkInterfaceList.json]: @dsgouda  but they are the part already from previous months... The only change we made is identity change... 
+ @lmazuel  could you please review what's the expected VMSS examples behavior here? As I remember you told me don't touch VMSS :)

Comment (lmazuel) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/examples/VmssVmNetworkInterfaceList.json]: @MikhailTryakhov @dsgouda The way VMSS in Network is designed, they have to copy the file **as-is** from one month to another. This implies copy the examples with it.
If we want to avoid that copy eventually, it requires a more deeper discussion on how we structure the Swagger for Network and the peculiar VMSS. I don't think that's the point of this PR, and shouldn't be address here.

Comment (MikhailTryakhov) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/examples/VmssVmNetworkInterfaceList.json]: @lmazuel so we should leave api version as is without api-version update? Agree ;)

Comment (dsgouda) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/examples/VmssVmNetworkInterfaceList.json]: @lmazuel agree but examples that do not match to the current version is a bit moot when being tested. Definitely needs a rethink.
@MikhailTryakhov please mark this as something we need to discuss in upcoming meetings.
Won't block on this

Comment (MikhailTryakhov) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/examples/VmssVmNetworkInterfaceList.json]: @dsgouda  done. please approve/merge this PR?

File: specification/network/resource-manager/readme.md [hunk 6]
```diff
@@ -731,6 +730,9 @@ output-folder: $(go-sdk-folder)/services/network/mgmt/2018-02-01/network
 ## Suppression
 ``` yaml
 directive:
+  - suppress: RequiredPropertiesMissingInResourceModel
```
Comment (dsgouda) [inline specification/network/resource-manager/readme.md]: Was this approved?

File: specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/virtualWan.json [hunk 7]
```diff
@@ -0,0 +1,2125 @@
+{
```
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/virtualWan.json]: @MikhailTryakhov Was this spec reviewed?

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks good other than a couple of comments.

**Comment by MikhailTryakhov** [comment]
@dsgouda all the tests successfully passed

File: specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/virtualWan.json (hunk 7 above)
Comment (MikhailTryakhov) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-04-01/virtualWan.json]: @dsgouda  https://github.com/Azure/azure-rest-api-specs/pull/3204

File: specification/network/resource-manager/readme.md (hunk 6 above)
Comment (MikhailTryakhov) [reply specification/network/resource-manager/readme.md]: yes, that is the same exception for ID readonly we had 1-2 months ago

**Comment by dsgouda** [comment]
@MikhailTryakhov Looks good, there is a model validation error [here](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/394786946) that needs a fix too
Apologize for the turnaround.

**Comment by MikhailTryakhov** [comment]
@dsgouda  I'm working with @yushwang for a 2nd month to make it fixed, now have 1 error instead of 3 in VnetGW and all of them fixed in express routes.
I'm afraid we can't wait for this error fix and have to publish with it. Fix would be merged to master when we'll have it ready

**Comment by dsgouda** [comment]
@MikhailTryakhov normally this would be a blocker but given the time crunch I will let this pass. Please make a note at your end to fix this ASAP

**Review by dsgouda — APPROVED** [review]

**Comment by dsgouda** [comment]
@lmazuel would it be okay to merge this, looks like travis is failing on timeouts while generating the sdks

**Comment by MikhailTryakhov** [comment]
@dsgouda added exception for this error approved by Gaurav (please see thread "VirtualNetworkGateway spec CR. Fixed 2 of 3 ARM errors")
So we've finally resolved it!

@lmazuel  could you please help with merge?

File: specification/network/resource-manager/readme.md [hunk 8]
```diff
@@ -31,8 +31,6 @@ openapi-type: arm
 tag: package-2018-04
 ```
 
-### Tag: package-2018-05
```
Comment (dsgouda) [inline specification/network/resource-manager/readme.md]: @MikhailTryakhov Please undo this change

**Comment by dsgouda** [comment]
@lmazuel Please merge since I do not have the permissions.

**Comment by dsgouda** [comment]
@marstr ping
